src/arch/arm/tlb.hh

```diff
@@ -156,9 +164,19 @@ class TLB : public BaseTLB
     int rangeMRU; //On lookup, only move entries ahead when outside rangeMRU
     vmid_t vmid;
 
+    /** Set of observed page sizes in the TLB
+     * We update the set conservatively, therefore allowing
+     * false positives but not false negatives.
+     * This means there could be a stored page size with
+     * no matching TLB entry (e.g. it has been invalidated),
+     * but if the page size is not in the set, we are certain
+     * there is no associated TLB with that size
+     */
+    std::set<Addr> observedPageSizes;
+
   public:
     using Params = ArmTLBParams;
-    using Lookup = TlbEntry::Lookup;
+    using Lookup = TlbEntry::KeyType;
```

Comment:
Nit: maybe keeping the Lookup alias increases the difficulty of learning the class

Reply:
This PR is already big and I wanted to make it less intrusive so that I didn't have to change all the Lookup instances (IIRC it is used a lot in the TLBI code). I can add an extra commit, alternatively I will open a new PR later on. Your choice

Reply:
It can be left for a future PR, no worries